### DISCUSSION TITLE
Generate xcarchive from a Maui project

### DISCUSSION THREAD

**Discussion by jerodmotley** [fastlane/fastlane#22287] [Q&A]
I have a .net Maui app that I'm trying to publish. Apple requires an xcarchive for the app to be uploaded but VS for Mac won't generate one anymore. It used to be generated when we used Xamarin but that has now been deprecated. Maui will successfully generate an ipa file but Apple won't accept that for upload. They want the xcarchive I'm not certain how to use Fastlane to generate the xcarchive without an xcodeproj file, which Maui also doesn't generate. Any thoughts?

**Comment by sonia598lewis** [comment]
Hello,
To generate an xcarchive [redhumana](https://www-redhumana.com) without an xcodeproj file using Fastlane, you can use the xcarchive action. Here's a simple command to get you started:
fastlane run xcarchive

You can pass additional parameters if needed, for example:
fastlane run xcarchive parameter1:"value1" parameter2:"value2"

Make sure you have Fastlane set up correctly and that your project is configured to use Fastlane. If you encounter any issues, the Fastlane documentation and community forums can be very helpful.

Best Regards,
Sonia Lewis
